webapp/src/js/services/database-connection-monitor.js

```diff
@@ -0,0 +1,28 @@
+/*
+PouchDB intermittently emits an unhandledrejection to ineffectively communicate that it has catastrophically failed.
+This service detects the event and resolves a promise when it occurs.
+
+DOMException: Failed to execute 'transaction' on 'IDBDatabase': The database connection is closing
+*/
+angular.module('inboxServices').factory('DatabaseConnectionMonitor',
+  function($window) {
+    'use strict';
+    'ngInject';
+
+    const promiseToCloseDatabase = new Promise(function(resolve) { 
+      $window.addEventListener('unhandledrejection', function(promiseRejectionEvent) {
+        if (
+          promiseRejectionEvent &&
+          promiseRejectionEvent.reason &&
+          promiseRejectionEvent.reason.message === 'Failed to execute \'transaction\' on \'IDBDatabase\': The database connection is closing.'
```

Comment:
I wonder if this is too restrictive - if they change the wording even a little then this will stop working, and we don't have any good way to add a test for regressions. Are there any other fields we can use (like an error number)? Maybe we should just check if the message contains "IDBDatabase"...

Reply:
Oh, I see you've added a unit test below that does exactly this! Nice one - ignore my comment.